File: 0036-valid-sudoku/0036-valid-sudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */
var isValidSudoku = function(board) {
    const rowSet = new Set(), colSet = new Set(), boxSet = new Set();
    
    for (let i = 0; i < board.length; i++) {
        for (let j = 0; j < board[i].length; j++) {
            let row = board[i][j];
            let col = board[j][i];
            let box = board[3 * Math.floor(i/3) + Math.floor(j / 3)][((i * 3) % 9) + (j % 3)]
            if (row !== ".") {
                if (colSet.has(row)) return false;
                colSet.add(row);
            }
             if (col !== ".") {
                if (rowSet.has(col)) return false;
                rowSet.add(col);
            }
             if (box !== ".") {
                if (boxSet.has(box)) return false;
                boxSet.add(box);
            }
            // console.log("rowset", rowSet)
            // console.log("colset", colSet)
        }    
        rowSet.clear();
        colSet.clear();
        boxSet.clear();
    }
    return true;
};

File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        rows = {}
        cols = {}
        box = {}
        
        
        for r in range(0, len(board)):
            for c in range(0, len(board[r])):
                current_val = board[r][c]
                
                if current_val == ".":
                    continue
                box_coord = f"{math.floor(r/3)}, {math.floor(c/3)}"
                
                if r not in rows:
                    rows[r] = set()
                if c not in cols:
                    cols[c] = set()
                if box_coord not in box:
                    box[box_coord] = set()
                    
                if current_val in rows[r] or current_val in cols[c] or current_val in box[box_coord]:
                    return False
                
                rows[r].add(current_val)
                cols[c].add(current_val)
                box[box_coord].add(current_val)
        
        return True